project/UI/Markdown/Markdown.swift

```diff
@@ -2,47 +2,41 @@
-public func markdown(script: String) -> Either<CLIKit.Console.Error, Void> {
+    public init() {}
     
-    func arguments(console: CLIKit.Console) -> IO<CLIKit.Console.Error, (input: URL, output: URL)> {
-        console.input().flatMap { args in
-            guard let inputPath = args[MarkdownCommand.project.rawValue]?.trimmingEmptyCharacters.expandingTildeInPath,
-                  let outputPath = args[MarkdownCommand.output.rawValue]?.trimmingEmptyCharacters.expandingTildeInPath else {
-                    return IO.raiseError(.arguments)
-            }
-            
-            let folder = URL(fileURLWithPath: inputPath, isDirectory: true)
-            let output = URL(fileURLWithPath: outputPath, isDirectory: true)
-
-            return IO.pure((input: folder, output: output))
-        }^
-    }
+    @ArgumentParser.Option(help: "Path to the folder containing Xcode Playground to render")
+    var project: String
+    
+    @ArgumentParser.Option(help: "Path where the resulting Markdown files will be generated")
+    var output: String
     
-    let console = Console(script: script,
-                          description: "Render markdown files from nef Playground",
-                          arguments: .init(name: MarkdownCommand.project.rawValue, placeholder: "path-to-input", description: "path to the folder containing Xcode Playground to render"),
-                                     .init(name: MarkdownCommand.output.rawValue, placeholder: "path-to-output", description: "path where the resulting Markdown files will be generated"))
+    var projectURL: URL { URL(fileURLWithPath: project.trimmingEmptyCharacters.expandingTildeInPath) }
+    var outputURL: URL  { URL(fileURLWithPath: output.trimmingEmptyCharacters.expandingTildeInPath) }
     
-    return arguments(console: console)
-        .flatMap { (input, output) in
-            nef.Markdown.render(playgroundsAt: input, into: output)
-               .provide(console)^
-               .mapError { _ in .render() }
-               .foldM({ _ in console.exit(failure: "rendering Xcode Playgrounds from '\(input.path)'") },
-                      { _ in console.exit(success: "rendered Xcode Playgrounds in '\(output.path)'")  }) }^
-        .reportStatus(in: console)
-        .foldM({ e in console.exit(failure: "\(e)")        },
-               { success in console.exit(success: success) })
-        .unsafeRunSyncEither()
+    
+    public func main() -> IO<CLIKit.Console.Error, Void> {
+        arguments(parsableCommand: self)
+            .flatMap { (input, output) in
+                nef.Markdown.render(playgroundsAt: input, into: output)
+                    .provide(Console.default)
+                    .mapError { _ in .render() }
+                    .foldM({ _ in Console.default.exit(failure: "rendering Xcode Playgrounds from '\(input.path)'") },
+                           { _ in Console.default.exit(success: "rendered Xcode Playgrounds in '\(output.path)'")   })
```

Comment:
Useless `mapError`.

Reply:
if I remove this useless ` .mapError { _ in .render() }` - it fails (looking for the type) but I will try again, maybe know I extracted this logic...

Reply:
The type may fail, what you need to wonder is if change the type is useful. The errors you are sending from the inner layer never get taken into consideration when mapping them in the outer layers.